diff --git a/Makefile b/Makefile
index 3448086..90c48ce 100644
--- a/Makefile
+++ b/Makefile
@@ -26,11 +26,6 @@
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 MY_PLEXIL_HOME := $(shell pwd)
-ifneq ($(PLEXIL_HOME),)
-ifneq ($(PLEXIL_HOME),$(MY_PLEXIL_HOME))
-$(error Environment variable PLEXIL_HOME is in error. It must be set to $(MY_PLEXIL_HOME) before proceeding)
-endif
-endif
 
 export PLEXIL_HOME := $(MY_PLEXIL_HOME)
 
@@ -62,6 +57,7 @@ essentials: universalExec TestExec IpcAdapter UdpAdapter plexil-compiler plexils
 # Standalone targets
 #
 
+.PHONY: checker pv plexil-compiler plexilscript robosim sample
 checker:
 	(cd checker && ant jar)
 
diff --git a/checker/build.xml b/checker/build.xml
index 6427f94..0c9e75d 100644
--- a/checker/build.xml
+++ b/checker/build.xml
@@ -13,14 +13,14 @@
   <property name = "domain-scope"       value = "main"/>
   <property name = "domain-scope-dir"   value = ""/>
   <property name = "main-class"         value = "DeclMain"/>
-  <property name = "lib-dir"            location = "${env.PLEXIL_HOME}/jars"/>
+  <property name = "jar-dir"            location = "/usr/share/java"/>
   <property name = "resources"          location = "."/>
   <property name = "compile-includes"   location = "reader/**,main/**,model/**"/>
   <property name = "compile-excludes"   location = ""/>
   
   <!-- java library files that belong in the class path and the jar -->
   
-  <fileset dir = "${lib-dir}" id = "libraries">
+  <fileset dir = "${jar-dir}" id = "libraries">
     <include name = "nanoxml.jar"/>
   </fileset>
   
diff --git a/compilers/plexil/Makefile b/compilers/plexil/Makefile
index 640177f..5f805e5 100644
--- a/compilers/plexil/Makefile
+++ b/compilers/plexil/Makefile
@@ -36,7 +36,7 @@ PARSER_JAVA_SRC_DIR	:= $(PARSER_JAVA_DIR)/plexil
 JAVA_SRCPATH		:= java
 JAVA_SRC_DIR		:= $(JAVA_SRCPATH)/plexil
 CLASSES_DIR			:= classes
-THIRD_PARTY_JAR_DIR	:= $(PLEXIL_HOME)/jars
+THIRD_PARTY_JAR_DIR	:= /usr/share/java
 
 # Files
 PARSER_SRCS		:= $(PARSER_SRC_DIR)/Plexil.g $(PARSER_SRC_DIR)/PlexilTreeTransforms.g
@@ -52,14 +52,14 @@ PLEXILC_JAR		:= $(PLEXIL_HOME)/jars/PlexilCompiler.jar
 DIRT			+= $(PARSER_JAVA) $(PARSER_CLASSES) $(JAVA_CLASSES)
 
 # Invoking ANTLR 3
-ANTLR3_TOOL_JAR	:= $(THIRD_PARTY_JAR_DIR)/antlr-3.5.2-complete-no-st3.jar
+ANTLR3_TOOL_JAR	:= $(THIRD_PARTY_JAR_DIR)/antlr3.jar
 ANTLR_OPTS		:= -fo $(PARSER_JAVA_SRC_DIR)
 ANTLR			:= $(JAVA) -jar $(ANTLR3_TOOL_JAR)
 
 # Other necessities
 ANTLR3_RUNTIME_JAR	:= $(ANTLR3_TOOL_JAR)
 NANOXML_JAR			:= $(THIRD_PARTY_JAR_DIR)/nanoxml.jar
-SAXON_JAR			:= $(THIRD_PARTY_JAR_DIR)/saxon.jar
+SAXON_JAR			:= $(THIRD_PARTY_JAR_DIR)/saxon/saxon.jar
 
 # Classpath for Java compiler
 JAVAC_CLASSPATH	:= $(ANTLR3_RUNTIME_JAR):$(NANOXML_JAR):$(SAXON_JAR):$(CLASSES_DIR)
diff --git a/compilers/plexil/PlexilCompiler b/compilers/plexil/PlexilCompiler
index b9db242..3a43d17 100755
--- a/compilers/plexil/PlexilCompiler
+++ b/compilers/plexil/PlexilCompiler
@@ -26,22 +26,8 @@
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
-# Check environment variables
-if [ ! $PLEXIL_HOME ]
-then
-    echo Error: Please set environment variable PLEXIL_HOME
-    echo to the full pathname of your 'plexil' or 'trunk' directory.
-    exit 1
-fi
+JAVA=${JAVA:-/usr/bin/java}
+JARS_DIR=/usr/share/java
+ANTLR_RUNTIME_JAR=antlr3-runtime.jar
 
-. ${PLEXIL_HOME}/scripts/plexil-setup.sh
-
-JARS_DIR=${PLEXIL_HOME}/jars
-ANTLR_RUNTIME_JAR=antlr-3.5.2-complete-no-st3.jar
-JAVA=java
-if [ -n "$JAVA_HOME" ]
-then
-    JAVA=$JAVA_HOME/bin/java
-fi
-
-exec $JAVA -classpath $JARS_DIR/PlexilCompiler.jar:$JARS_DIR/$ANTLR_RUNTIME_JAR:$JARS_DIR/nanoxml.jar:$JARS_DIR/saxon.jar plexil.Compiler $@
+exec $JAVA -classpath $JARS_DIR/PlexilCompiler.jar:$JARS_DIR/$ANTLR_RUNTIME_JAR:$JARS_DIR/nanoxml.jar:$JARS_DIR/saxon/saxon.jar plexil.Compiler $@
diff --git a/compilers/plexil/PlexilCompilerDebug b/compilers/plexil/PlexilCompilerDebug
index 3f3c06a..59574a8 100755
--- a/compilers/plexil/PlexilCompilerDebug
+++ b/compilers/plexil/PlexilCompilerDebug
@@ -25,23 +25,4 @@
 # TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-
-# Check environment variables
-if [ ! $PLEXIL_HOME ]
-then
-    echo Error: Please set environment variable PLEXIL_HOME
-    echo to the full pathname of your 'plexil' or 'trunk' directory.
-    exit 1
-fi
-
-. ${PLEXIL_HOME}/scripts/plexil-setup.sh
-
-JARS_DIR=${PLEXIL_HOME}/jars
-ANTLR_RUNTIME_JAR=antlr-3.5.2-complete-no-st3.jar
-JDB=jdb
-if [ -n "$JAVA_HOME" ]
-then
-    JDB=$JAVA_HOME/bin/jdb
-fi
-
-exec $JDB -classpath $JARS_DIR/PlexilCompiler.jar:$JARS_DIR/$ANTLR_RUNTIME_JAR:$JARS_DIR/nanoxml.jar:$JARS_DIR/saxon.jar plexil.Compiler $@
+JAVA=/usr/bin/jdb ./PlexilCompiler $@
diff --git a/compilers/plexilisp/emacs.el b/compilers/plexilisp/emacs.el
index 807006b..e818873 100644
--- a/compilers/plexilisp/emacs.el
+++ b/compilers/plexilisp/emacs.el
@@ -25,7 +25,7 @@
 
 (defun init-plexilisp (&optional compile)
   (interactive)
-  (let* ((plexildir (format "%s/compilers/plexilisp" (getenv "PLEXIL_HOME")))
+  (let* ((plexildir (format "%s/plexilisp" (getenv "PLEXIL_HOME")))
          (source-file (format "%s/plexil.el" plexildir))
          (compiled-file (format "%s/plexil.elc" plexildir))
          (max-specpdl-size 2700))
diff --git a/compilers/plexilisp/plexil.el b/compilers/plexilisp/plexil.el
index 0ec7427..c136ce6 100644
--- a/compilers/plexilisp/plexil.el
+++ b/compilers/plexilisp/plexil.el
@@ -53,7 +53,7 @@
 ;;;   ...the rest should be intuitive
 
 (defconst *plexil-home* (format "%s" (getenv "PLEXIL_HOME")))
-(defconst *plexilisp-location* (format "%s/compilers/plexilisp" *plexil-home*))
+(defconst *plexilisp-location* (format "%s/plexilisp" *plexil-home*))
 (defconst *bin-location* (format "%s/scripts" *plexil-home*))
 
 ;;; Note that this version is for Plexilisp only; not tied to
diff --git a/compilers/plexilscript/build.xml b/compilers/plexilscript/build.xml
index 56fe9d0..159d51c 100644
--- a/compilers/plexilscript/build.xml
+++ b/compilers/plexilscript/build.xml
@@ -5,13 +5,13 @@
   </description>
 
   <property environment="env"/>
-  <property name="dir.antlr" location="${env.PLEXIL_HOME}/jars"/>
+  <property name="dir.antlr" location="/usr/share/java"/>
   <property name="dir.grammars" location="antlr"/>
   <property name="dir.java" location="java"/>
   <property name="dir.generated" location="generated"/>
   <property name="dir.classes" location="classes"/>
   <property name="dir.jars" location="jars"/>
-  <property name="dir.install" location="${env.PLEXIL_HOME}"/>
+  <property name="dir.install" location="../../jars"/>
 
   <path id="antlr.classpath">
       <pathelement path="${classpath}"/>
@@ -32,7 +32,7 @@
 
   <target name="install" depends="plexilscript-jar">
     <copy file="${dir.jars}/plexilscript.jar"
-	  todir="${env.PLEXIL_HOME}/jars"
+      todir="${dir.install}"
 	  preservelastmodified="true"/>
   </target>
 
diff --git a/scripts/checkPlexil b/scripts/checkPlexil
index c313859..a8fdd71 100755
--- a/scripts/checkPlexil
+++ b/scripts/checkPlexil
@@ -42,21 +42,14 @@ then
     exit 2
 fi
 
-prog=${PLEXIL_HOME}/checker/global-decl-checker.jar
+prog=/usr/share/java/plexil/global-decl-checker.jar
 args=$*
 JAVA=java
-if [ -n "$JAVA_HOME" ]
-then
-    JAVA=$JAVA_HOME/bin/java
-fi
 
 # Make sure jar file exists
 if [ ! -f "${prog}" ]
 then
-  echo "Cannot find file ${prog}. To build it:"
-  echo "  cd ${PLEXIL_HOME}/checker"
-  echo "  ant jar"
-  echo "If JAR is located elsewhere, please update the location in the script."
+  echo "Cannot find file ${prog}."
   exit 2
 fi
 
diff --git a/scripts/plexil.profile.sh b/scripts/plexil.profile.sh
new file mode 100644
index 0000000..7f90c6c
--- /dev/null
+++ b/scripts/plexil.profile.sh
@@ -0,0 +1,5 @@
+# /etc/profile.d/plexil.sh - set PLEXIL_HOME if not set yet
+
+PLEXIL_HOME=${PLEXIL_HOME:-/usr/share/plexil}
+
+export PLEXIL_HOME
diff --git a/scripts/plexilc b/scripts/plexilc
index 1c73bb3..e9c9e32 100755
--- a/scripts/plexilc
+++ b/scripts/plexilc
@@ -62,19 +62,13 @@ then
     exit 1
 fi
 
-. ${PLEXIL_HOME}/scripts/plexil-setup.sh
-
-JARS_DIR=${PLEXIL_HOME}/jars
-ANTLR_RUNTIME_JAR=antlr-3.5.2-complete-no-st3.jar
+JARS_DIR=/usr/share/java/
+ANTLR_RUNTIME_JAR=antlr3-runtime.jar
 JAVA=java
-if [ -n "$JAVA_HOME" ]
-then
-    JAVA=$JAVA_HOME/bin/java
-fi
 
 # Commands to execute
-plexil_compiler="$JAVA -classpath $JARS_DIR/PlexilCompiler.jar:$JARS_DIR/$ANTLR_RUNTIME_JAR:$JARS_DIR/nanoxml.jar:$JARS_DIR/saxon.jar plexil.Compiler"
-plexilscript_compiler="$JAVA -cp ${JARS_DIR}/plexilscript.jar:${JARS_DIR}/antlr.jar plexilscript.PlexilScriptParse"
+plexil_compiler="$JAVA -classpath $JARS_DIR/PlexilCompiler.jar:$JARS_DIR/$ANTLR_RUNTIME_JAR:$JARS_DIR/nanoxml.jar:$JARS_DIR/saxon/saxon.jar plexil.Compiler"
+plexilscript_compiler="$JAVA -cp ${JARS_DIR}/plexilscript.jar:${JARS_DIR}/antlr3.jar plexilscript.PlexilScriptParse"
 plexil_checker="$PLEXIL_HOME/scripts/checkDecls"
 
 # Parse options
@@ -172,9 +166,9 @@ then
 
     if [ -n "$quiet" ]
     then
-	emacs --batch --no-site-file --load $PLEXIL_HOME/compilers/plexilisp/plexil.el $1 --eval '(plexil)' > /dev/null 2>&1
+	emacs --batch --no-site-file --load $PLEXIL_HOME/plexilisp/plexil.el $1 --eval '(plexil)' > /dev/null 2>&1
     else
-	emacs --batch --no-site-file --load $PLEXIL_HOME/compilers/plexilisp/plexil.el $1 --eval '(plexil)'
+	emacs --batch --no-site-file --load $PLEXIL_HOME/plexilisp/plexil.el $1 --eval '(plexil)'
     fi
 
 	# If plexilisp generates a plan, we can check it.
diff --git a/scripts/plexilisp b/scripts/plexilisp
index f211fe1..e311b00 100755
--- a/scripts/plexilisp
+++ b/scripts/plexilisp
@@ -37,4 +37,4 @@ then
     exit 1
 fi
 
-emacs --batch --no-site-file --load $PLEXIL_HOME/compilers/plexilisp/plexil.el $1 --eval '(plexil)'
+emacs --batch --no-site-file --load $PLEXIL_HOME/plexilisp/plexil.el $1 --eval '(plexil)'
